util/set/slice.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package set
```

Comment:
needs package doc. maybe in a new file `set.go` if we foresee more types of sets. (I think we have other types in corp already?)

Reply:
Package doc already exists in `set.go`, albeit not the most verbose:
https://github.com/tailscale/tailscale/blob/main/util/set/set.go#L4